Tribler/Test/Core/Upgrade/test_upgrader.py

```diff
@@ -54,10 +41,40 @@ def test_upgrade_72_to_pony(self):
         old_database_path = os.path.join(self.session.config.get_state_dir(), 'sqlite', 'tribler.sdb')
         new_database_path = os.path.join(self.session.config.get_state_dir(), 'sqlite', 'metadata.db')
         channels_dir = os.path.join(self.session.config.get_chant_channels_dir())
+
         shutil.copyfile(OLD_DB_SAMPLE, old_database_path)
-        self.upgrader.upgrade_72_to_pony()
-        yield self.upgrader.finished_deferred
+        yield self.upgrader.run()
         mds = MetadataStore(new_database_path, channels_dir, self.session.trustchain_keypair)
         with db_session:
             self.assertEqual(mds.TorrentMetadata.select().count(), 24)
         mds.shutdown()
+
+    @trial_timeout(10)
+    @inlineCallbacks
+    def test_skip_upgrade_72_to_pony(self):
+        OLD_DB_SAMPLE = os.path.abspath(os.path.join(os.path.abspath(
+            os.path.dirname(os.path.realpath(__file__))), '..', 'data', 'upgrade_databases', 'tribler_v29.sdb'))
+        old_database_path = os.path.join(self.session.config.get_state_dir(), 'sqlite', 'tribler.sdb')
+        new_database_path = os.path.join(self.session.config.get_state_dir(), 'sqlite', 'metadata.db')
+        channels_dir = os.path.join(self.session.config.get_chant_channels_dir())
+
+        shutil.copyfile(OLD_DB_SAMPLE, old_database_path)
+
+        from Tribler.Core.Upgrade.db72_to_pony import DispersyToPonyMigration
+
+        original_init = DispersyToPonyMigration.__init__
```

Comment:
This is _very_ dangerous to do. If the test fails for some reason after this line, and before you replace the init method with the old one, the `DispersyToPonyMigration` initializer will be stuck with the patched method and might fail in subsequent tests. We had these kind of errors in the past and they are a headache to debug.

If possible, try to avoid monkey patching methods on raw classes.

Reply:
I used monkey patching because there are only a couple of tests in the file, basically testing the same thing. The problem with this kind of stuff is that the alternative is to introduce more backreferences, which we try to avoid. I'll see what I can do, but I would leave it as it is, instead of adding backreference to e.g. `session` in `DispersyToPonyMigration`. "Backreference" often means "stuck memory".